communitiesuk/funding-service-design-post-award-data-store
# Celery setup

Set up celery for asynchronous download task.

Ticket:
https://dluhcdigital.atlassian.net/browse/FPASF-102


### How to test
When running unit tests, we've enabled `tasks_always_eager` on celery, which means that any calls to `.delay` are effectively ignored. Tasks will always run synchronously. This makes it easier to test the behaviour of the task is as we expect.

It does mean that we aren't testing anything about celery's queueing logic or that we have any workers configured and running correctly, but testing that would add a lot of overhead and complexity.

We should consider adding some of these functional/full integration tests in the future, probably running in deployed environments (eg test environment) before releasing to production.
